legate/core/operation.py

```diff
@@ -189,32 +216,40 @@ def declare_partition(self, store, disjoint=True, complete=True):
 
 
 class Task(Operation):
-    def __init__(self, context, task_id, mapper_id=0, op_id=0):
-        Operation.__init__(self, context, mapper_id=mapper_id, op_id=op_id)
+    def __init__(
+        self,
+        context: Context,
+        task_id: int,
+        mapper_id: int = 0,
+        op_id: int = 0,
+    ) -> None:
+        super().__init__(context, mapper_id=mapper_id, op_id=op_id)
         self._task_id = task_id
-        self._scalar_args = []
-        self._comm_args = []
+        self._scalar_args: list[tuple[Any, Any]] = []
+        self._comm_args: list[Communicator] = []
 
     @property
-    def uses_communicator(self):
+    def uses_communicator(self) -> bool:
         return len(self._comm_args) > 0
 
-    def get_name(self):
+    def get_name(self) -> str:
         libname = self.context.library.get_name()
         return f"{libname}.Task(tid:{self._task_id}, uid:{self._op_id})"
 
-    def add_scalar_arg(self, value, dtype):
+    def add_scalar_arg(self, value: Any, dtype: Any) -> None:
```

Comment:
again, `dtype`'s type could be `DTType`.

Reply:
https://github.com/nv-legate/legate.core/pull/189/commits/662eda7182a8c5496167277d8bc8490b98736306